woocommerce/woocommerce
# Use `createRoot().render()` instead of `render()`

### Changes proposed in this Pull Request:

This PR introduces the following changes:

1. **Replacement of `ReactDOM.render`:** The deprecated `ReactDOM.render` has been replaced with `createRoot` from `@wordpress/element`. This transition aligns with React 18's requirements for component rendering.

2. **Use of React 18 Features:** The adoption of `createRoot` enables the utilisation of React 18's advanced features, like concurrent mode.

Closes #48797.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Smoke-test the Cart and Checkout blocks, including the Order Confirmation template.
2. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`.
3. Add a new pickup location.
4. Verify that the pickup location can be saved.
5. Ensure that all e2e tests are passing.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Switch `render()` to `createRoot().render()` to use React 18 features.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
